src/Action/Search.hs

```diff
@@ -33,10 +33,18 @@ actionSearch Search{..} = replicateM_ repeat_ $ -- deliberately reopen the datab
         whenLoud $ putStrLn $ "Query: " ++ unescapeHTML (renderQuery q)
         let (shown, hidden) = splitAt count $ nubOrd $ map targetItem res
         hSetEncoding stdout utf8
-        putStr $ unlines $ map (unescapeHTML . innerTextHTML) shown
+        let toShow = if whenCounter numbers info then addCounter shown else shown
+        putStr $ unlines $ map (unescapeHTML . innerTextHTML) toShow
         when (hidden /= []) $ do
             whenNormal $ putStrLn $ "-- plus more results not shown, pass --count=" ++ show (count+10) ++ " to see more"
 
+addCounter :: [String] -> [String]
+addCounter = addCounter' 1
+    where addCounter' num (x:xs) = (show num ++ ") " ++ x) :  (addCounter' (num + 1) xs)
+          addCounter' _ [] = []
```

Comment:
I think this would look nicer as a zip/zipWith, something like:

    addCounter = zipWith (\i x -> show i ++ ") " ++ x) [1..]